frontend/src/stores/redux/loggedUser/loggedUserSlice.ts

```diff
@@ -0,0 +1,36 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit'
+import {RootState} from "../store";
+import {ILoggedUserState} from "../../../types/ILoggedUserState";
+import {ILoggedUser} from "./loggedUser";
+import {UserType} from "../../../types/UserType";
+
+const initialState: ILoggedUserState = {
+    loggedUserState: {
+        id: -1,
+        email: "",
+        username: "",
+        userType: UserType.Customer,
+        token: ""
+    }
+}
+
+export const loggedUserSlice = createSlice({
+    name: 'loggedUser',
+    initialState,
+    reducers: {
+        signIn: (state, action: PayloadAction<ILoggedUser>) => {
+            state.loggedUserState = action.payload;
```

Comment:
it's not secure assign a pointer to a action object, because object itself can be changed later somewhere. You have to create a new object bases on action.payload and then assign it to a state.
`state.loggedUserState = {...action.payload}`